crypto/thread/api.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019-2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/configuration.h>
+#include <openssl/thread.h>
+#include <internal/thread.h>
+
+uint32_t OSSL_get_thread_support_flags(void)
+{
+    int support = 0;
+
+#if defined(OPENSSL_THREAD_POOL)
+    support |= OSSL_THREAD_SUPPORT_FLAG_THREAD_POOL;
+#endif
+#if defined(OPENSSL_DEFAULT_THREAD_POOL)
+    support |= OSSL_THREAD_SUPPORT_FLAG_DEFAULT_SPAWN;
+#endif
+
+    return support;
+}
+
+#if !defined(OPENSSL_THREAD_POOL) && !defined(OPENSSL_DEFAULT_THREAD_POOL)
+
+int OSSL_set_max_threads(uint64_t max_threads)
+{
+    return 0;
+}
+
+uint64_t OSSL_get_max_threads(void)
+{
+    return 0;
+}
+
+#else
+
+uint64_t OSSL_get_max_threads(void)
+{
+    uint64_t ret = 0;
+    OSSL_LIB_CTX_THREADS *tdata = OSSL_LIB_CTX_GET_THREADS(NULL);
```

Comment:
Should we be passing a libctx here?

Reply:
Please see https://github.com/openssl/openssl/pull/12255/#discussion_r984669150

Reply:
See also https://github.com/openssl/openssl/pull/18864#issuecomment-1263359384